Js dom api task

maksimsergeev97 opened the pull request:
### js-dom-api

[Demo](https://maksimsergeev97.github.io/JS-DOM-Practice) |
[Code base](https://github.com/maksimsergeev97/JS-DOM-Practice)

The code is submitted in a dedicated feature branch.

Only code files are submitted.

Please, review.

maksimsergeev97 commented:
Thank you! in the students' chat, they pointed out to me errors related to styles, I corrected them. I went through typical errors, I didn’t find such in my code. Please review :)

madmaxWMFU commented on submissions/maksimsergeev97/js-dom/script.js:
```diff
@@ -0,0 +1,56 @@
+  
+    function showMenuBurger() {
+      menuBurger.classList.add("menu__burger__active");
+      menuBtn.style.display = "none";
```
Better used `Element.classList.add/remove/toggle`

madmaxWMFU commented on submissions/maksimsergeev97/js-dom/script.js:
```diff
@@ -0,0 +1,56 @@
+      menuBtn.style.display = "block";
+    }
+  
+    menuBtn.addEventListener("click", () => {
+      showMenuBurger();
+    });
```
```suggestion
    menuBtn.addEventListener("click", showMenuBurger);
```

madmaxWMFU commented on submissions/maksimsergeev97/js-dom/script.js:
```diff
@@ -0,0 +1,56 @@
+      const mainBlock = document.querySelector(".main");
+      links.forEach((item, i) => {
+        item.addEventListener("click", () => {
+          fetch("js/base.json")
+            .then((response) => {
+              return response.json();
+            })
+            .then((data) => {
+              changeContent(data[i]);
+            });
```
Create a separate function that has the name `getData()` and do error catches.

madmaxWMFU commented on submissions/maksimsergeev97/js-dom/script.js:
```diff
@@ -0,0 +1,56 @@
+      });
+    }
+  
+    addEventForItem(".menu__item");
+    addEventForItem(".menu__burger__item");
```
Read about [event delegation](https://javascript.info/event-delegation).
```javascript
const burgerMenu = ;
const mainMenu = ;
const renderContent = ({ targer }) => {}
burgerMenu.addEventListener('click', renderContent);
mainMenu.addEventListener('click', renderContent);
```

madmaxWMFU commented on submissions/maksimsergeev97/js-dom/script.js:
```diff
@@ -0,0 +1,56 @@
+  
+
+
+  
```
Add an empty line.

madmaxWMFU reviewed (changes requested):
Good work. See some comments below.

maksimsergeev97 commented:
Hi! Thanks for the feedback! 
I fixed all bugs:
- removed inline styles and added classList;
- rewrote the event handler construction as you said;
- added new function getData() and catch errors;
- rewrote event handler assignments using event delegation. Hope it's correct now :)
- added empty line.
Please review, again.

madmaxWMFU commented on submissions/maksimsergeev97/js-dom/script.js:
```diff
@@ -0,0 +1,68 @@
+                                 <p class="descr">${item.descr}</p>`;
+    }
+
+    function getData (item) {
+      fetch("js/base.json")
+            .then((response) => {
+              return response.json();
+            })
+            .then((data) => {
+              changeContent(data[item]);
+            }).catch((error) => {
+              console.log(error);
+            })
+    }
```
Get once fetching data to the variable and work with it.
Then use only methods `Array.find()` or `Array.filter()`.

madmaxWMFU commented on submissions/maksimsergeev97/js-dom/script.js:
```diff
@@ -0,0 +1,68 @@
+            })
+            .then((data) => {
+              changeContent(data[item]);
+            }).catch((error) => {
+              console.log(error);
+            })
```
If was an error user did not see it.

madmaxWMFU commented on submissions/maksimsergeev97/js-dom/script.js:
```diff
@@ -0,0 +1,68 @@
+            })
+    }
+
+    function renderContent (selector, target) {
+      const items = document.querySelectorAll(selector);
+      items.forEach((item,i) => {
+        if (target == item && target.tagName == "LI") {
+          getData(i);
+          hideMenuBurger();
+        } else {
+          return;
+        };
+      });
+    };
+
+    function addEventClick (menuSelector, itemsSelector) {
+      const menu = document.querySelector(menuSelector)
+      menu.addEventListener('click', (e) => {
+        const target = e.target;
+        renderContent(itemsSelector, target);
+      });
+    }
+
+    addEventClick(".menu__list", ".menu__item");
+    addEventClick(".menu__burger__list", ".menu__burger__item")
```
```suggestion
    const menu = document.querySelector(".menu__list");
    const burgerMenu = document.querySelector(".menu__burger__list");
     
     function renderContent ({ target }) {
        if (target.classList. contains('menu__item') || target.classList. contains('menu__burger__item') ) {
        const targetData = fetchData.find(item => target.innerText.toLowerCase() === item.title);
          changeContent(targetData)
          hideMenuBurger();
        } 
    };  
     menu.addEventListener('click', renderContent);
     menu.addEventListener('click', burgerMenu);
```

If you realize once get data from base.json 

Read about [destruction](https://javascript.info/destructuring-assignment) to understand a construction `{ target }`.
The variable `fetchData` contains your data from base.json. If you implement one-time data loading.

madmaxWMFU commented on submissions/maksimsergeev97/js-dom/script.js:
```diff
@@ -0,0 +1,68 @@
+
+
+
+  
```
Add an empty line.
